### PR TITLE
Move PawnlessFlanks from Evaluation to Pawns.

### PR DESCRIPTION
**Pull request by protonspring**
This is not a classical simplification.  It does not reduce the code, but reduces the instructions performed during execution.  On my count in bench, PawnlessFlank is applied about 5 million times in evaluate.cpp, but is only applied 1.5 million times in pawns. The bench is not the same because PawnlessFlanks is now part of the KingDanger metric (which is probably where it should be as well).  Hence, the slightly adjusted score.

It unwinds some convoluted math and moves PawnlessFlank to pawns where it belongs, because it only depends on pawns and king.  See locutus2's comment below which is very articulate.

STC: 
LLR: 2.95 (-2.94,2.94) [-3.00,1.00]
Total: 55340 W: 11151 L: 11093 D: 33096
http://tests.stockfishchess.org/tests/view/5ad18d120ebc594bca9c0a06

LTC: 
LLR: 2.96 (-2.94,2.94) [-3.00,1.00]
Total: 28990 W: 4342 L: 4233 D: 20415
http://tests.stockfishchess.org/tests/view/5ad1975a0ebc594bca9c0a0a

Bench: 4708980

### CODE REVIEW

**Comment by Hanamuke** [comment]
Maybe adding a bench to show the speedup would be a good start

**Comment by protonspring** [comment]
Is there a way to show performance speed when the Bench value is changed?

**Comment by protonspring** [comment]
Because Tomtor suggested 0,4 and there were questions regarding performance, I decided to run an 0,4 test on the framework.  My apologies to @Rocky640 for being hard to follow.  I just though it would answer some questions.  It passed yellow indicating a 0.7 elo gain.

LLR: -2.95 (-2.94,2.94) [0.00,4.00]
Total: 83281 W: 12411 L: 12319 D: 58551
http://tests.stockfishchess.org/tests/view/5ad28de50ebc597834125699

This seems to suggest that this quazi-simplification is either it IS a bit faster, or the updated penalty value is a bit better.  Either way, it seems clear enough (at least to me) that this is just a bit better than master, but not by much.

**Comment by snicolet** [comment]
The [0..4] were the right bounds for this kind of patch, as this is what has been used in the past for similar refactoring when the bench changes or when we expect a speed gain. It is a little bit of a pity that the test failed, because there have been quite a few attempts to unify the unify the Pawnless flank penalty and the minKingPawnDistance penalty.

My advice would be to take the result positively: my feeling is that there may be an excellent opportunity to get a clean green SPRT[0..4] by keeping the structure of your best attempts and tweaking the bonus values a little bit. Let's be ambitious and hunt an Elo gain!

**Comment by protonspring** [comment]
Great.  Thx for the comments.  We won't be able to get the same bench without some intrusive mathematical adjustments which would not be pretty.  Also, I did LOTS of tuning so I'm not sure more will help.  I will try a few more tests, but other than guessing a new values, I don't know what more I could do.

@Rocky640 yes. . my bad.  I will fix that as well.  Thx.

**Comment by snicolet** [comment]
I am closing this pull request for now because of the bug noted by Alain (adding a Score to a Value), and also because each commit added to branch sends an email to all the participants of the thread (not sure about the fact that closing the pull request will stop this spamming, though... maybe it would be better to start anew and make the tests from another, clean branch?)

**Comment by protonspring** [comment]
Ok.  If i can get it to pass 0,4 i will retry.

**Comment by protonspring** [comment]
My last peace: this was a conceptual simplification and was never meant to pass 0,4.  I'm sad that it won't be applied and I think it's a shame.  It's theoretically faster (by a little bit) because it applies the bonus far fewer times and it is conceptually more simple because such a pawn bonus/penalty is now done in pawns only.  Also, this patch beat master on every test on the framework (STC and LTC).

That's all.  I'm moving on to something else.
